unified_planning/io/python_writer.py

```diff
@@ -276,6 +290,48 @@ def _write_problem_code(self, out: IO[str]):
                 raise NotImplementedError
             out.write(')\n')
 
+        if self.problem.kind.has_hierarchical():  # type: ignore
+            assert isinstance(self.problem, up.model.htn.hierarchical_problem.HierarchicalProblem)  # type: ignore
```

Comment:
We removed almost all `#type: ignore` and we are trying to avoid them.

For the first one, we disabled the type checking for the `ProblemKind` class.

Reply:
That's exactly what I tried but that was before rebasing. So it was probably be fixed in the commit added with the rebase.